test/benchmark_http_client_test.cc

```diff
@@ -276,8 +297,38 @@ TEST_F(BenchmarkClientHttpTest, BadContentLength) {
   };
 
   EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(0);
-  testBasicFunctionality(1, 1, 1);
+  testBasicFunctionality(1, 1, 1, request_generator);
   EXPECT_EQ(1, getCounter("http_2xx"));
 }
-
+TEST_F(BenchmarkClientHttpTest, MultipleRequestsDifferentPath) {
+  std::vector<HeaderMapPtr> requests_vector;
+  std::initializer_list<std::pair<std::string, std::string>> header1{{":scheme", "http"},
+                                                                     {":method", "GET"},
+                                                                     {":path", "/a"},
+                                                                     {":host", "localhost"},
+                                                                     {"Content-Length", "1313"}};
+  std::initializer_list<std::pair<std::string, std::string>> header2{{":scheme", "http"},
+                                                                     {":method", "GET"},
+                                                                     {":path", "/b"},
+                                                                     {":host", "localhost"},
+                                                                     {"Content-Length", "1313"}};
+  requests_vector.push_back(std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header1));
+  requests_vector.push_back(std::make_shared<Envoy::Http::TestRequestHeaderMapImpl>(header2));
+  std::vector<HeaderMapPtr>::iterator request_iterator;
+  request_iterator = requests_vector.begin();
+  RequestGenerator request_generator = [&]() {
+    auto item = *request_iterator;
+    request_iterator++;
+    return std::make_unique<RequestImpl>(item);
+  };
+  std::vector<absl::flat_hash_map<std::string, std::string>> expected_requests_vector;
+  expected_requests_vector.push_back(
+      getTestRecordedProperties(Envoy::Http::TestRequestHeaderMapImpl(header1)));
+  expected_requests_vector.push_back(
+      getTestRecordedProperties(Envoy::Http::TestRequestHeaderMapImpl(header2)));
+
+  EXPECT_CALL(stream_encoder_, encodeData(_, _)).Times(2);
+  testBasicFunctionality(1, 1, 2, request_generator, expected_requests_vector);
```

Comment:
some comment that explains that the in and out headers check will be done by the testBasicFunctionality which is the main test case.

Reply:
Done.